truss-chains/truss_chains/remote.py

```diff
@@ -436,7 +358,68 @@ def push(
     non_entrypoint_root_dir: Optional[str] = None,
     gen_root: pathlib.Path = pathlib.Path(tempfile.gettempdir()),
 ) -> Optional[ChainService]:
-    return _Pusher(options, gen_root).push(entrypoint, non_entrypoint_root_dir)
+    chainlet_artifacts = _ChainSourceGenerator(
+        options, gen_root
+    ).generate_chainlet_artifacts(
+        entrypoint,
+        non_entrypoint_root_dir,
+    )
+
+    if options.only_generate_trusses:
+        return None
+
+    if isinstance(options, definitions.PushOptionsBaseten):
+        _create_chains_secret_if_missing(options.remote_provider)
+        return _create_baseten_chain(options, chainlet_artifacts)
+    elif isinstance(options, definitions.PushOptionsLocalDocker):
+        entrypoint_service: Optional[DockerTrussService] = None
+        chainlet_to_predict_url: Dict[str, Dict[str, str]] = {}
+        chainlet_to_port: Dict[str, int] = {}
+
+        for chainlet_artifact in chainlet_artifacts:
+            port = utils.get_free_port()
+            chainlet_to_port[chainlet_artifact.name] = port
+
+            service = DockerTrussService(
+                # http://localhost:{port} seems to only work *sometimes* with docker.
+                f"http://host.docker.internal:{port}",
+                is_draft=True,
+            )
+
+            chainlet_to_predict_url[chainlet_artifact.name] = {
+                "predict_url": service.predict_url,
+            }
+
+            if chainlet_artifact.is_entrypoint:
+                assert entrypoint_service is None
+
+                entrypoint_service = service
+
+        assert entrypoint_service is not None
+
+        local_config_handler.LocalConfigHandler.set_dynamic_config(
+            definitions.DYNAMIC_CHAINLET_CONFIG_KEY,
+            json.dumps(chainlet_to_predict_url),
+        )
+
+        for chainlet_artifact in chainlet_artifacts:
```

Comment:
It's a bit of a nit, but I stumbled over why there are two loop, and how `service = DockerTrussService` is not already enough and a second step `_push_service_docker` is needed.

I think the current structure comes from first creating *all* entries in `chainlet_to_predict_url`...

I think it would be cool to rework this a bit. But I want to get this PR out, so please just leave a TODO note.

Reply:
Sounds good. Maybe I'll take a stab at it at some point. Let's just proceed now

Reply:
If we used separate config paths under `/tmp` that might work. The current logic feels a little fragile because the config path is not only shared between containers in one Chain, but across all Chains deployed using local Docker.